.github/workflows/publish_snapshot.yml

```diff
@@ -37,7 +37,7 @@ jobs:
           - branch-3.1
     steps:
     - name: Checkout Spark repository
-      uses: actions/checkout@61b9e3751b92087fd0b06925ba6dd6314e06f089 # pin@master
+      uses: actions/checkout@v3
```

Comment:
I also take a look on https://github.com/apache/spark/commit/00b87c967ff8217b64e597400f3248c375a74879, I think pin SHA is only for 3rd party, but for `actions/` I think tag is enough.